goc/bin/copy_config_osg

```diff
@@ -18,21 +18,105 @@ if [ ! -f /etc/cvmfs/repositories.d/config-osg.opensciencegrid.org/client.conf ]
     exit 1
 fi
 
-set -ex
-mkdir -p /cvmfs/config-osg-itb.opensciencegrid.org /etc/cvmfs/config.d
-cat >/etc/cvmfs/config.d/config-osg.opensciencegrid.org.local <<!EOF!
-CVMFS_SERVER_URL=http://oasis-itb.opensciencegrid.org:8000/cvmfs/@fqrn@
-CVMFS_PUBLIC_KEY=/etc/cvmfs/keys/opensciencegrid.org.pub:/srv/etc/cvmfs/oasis-itb.opensciencegrid.org.pub
-CVMFS_HTTP_PROXY=DIRECT
-unset CVMFS_CONFIG_REPOSITORY
-!EOF!
-mount -t cvmfs config-osg.opensciencegrid.org /cvmfs/config-osg-itb.opensciencegrid.org
-if ! cvmfs_server transaction config-osg.opensciencegrid.org; then
-    cvmfs_server abort -f config-osg.opensciencegrid.org
-    cvmfs_server transaction config-osg.opensciencegrid.org
+get_branch()
+{
+    # get branch name
+    raw_current_branch=`git branch | grep "*"`
+    current_branch=${raw_current_branch#* }
+    # get last component of branch name
+    current_branch_name=${current_branch##*\/}
+}
+
+confirm() 
+{
+    # call with a prompt string or use a default
+    read -r -p "${1:-Are you sure? [y/N]} " response
+    case "$response" in
+        [yY][eE][sS]|[yY]) 
+            false
+            ;;
+        *)
+            true
+            ;;
+    esac
+}
+
+# run only on -itb host 
+if [[ $(hostname -s) = *-itb ]]; then
+    # run only if non-root
+    if ! [ $(id -u) = 0 ]; then
+        if git remote -v 2>/dev/null | grep -q /config-repo.git; then
+            current_dir="$(git rev-parse --show-toplevel)"
+            cd ${current_dir}
+            git pull upstream osg
```

Comment:
Getting close.  Still waiting on the confirmation that is in the osg branch.

Reply:
`git pull upstream osg` will not fail if the current working branch is not osg, it will just merge the upstream osg branch into whatever the current branch is.  Yes there should be a check for current_branch_name == osg, else fail.  (After that the developer should normally create a new branch before committing the changes and making a PR).

Reply:
OK, makes sense now, wasn't sure how it behaves, should have tested myself and see, though.